File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let dict = {}
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] in dict) dict[nums[i]]+=1;
        else dict[nums[i]] = 1;

        if (dict[nums[i]] >= Math.ceil(nums.length/2)) {
            return nums[i]
        };
    };
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        mapper = {}

        for i in range(len(nums)):
            if nums[i] not in mapper:
                mapper[nums[i]] = 1
            else:
                mapper[nums[i]]+=1

            if mapper[nums[i]] >= math.ceil(len(nums)/2):
                p1 = nums[i]
                return p1
